plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeConfig.java

```diff
@@ -475,4 +487,53 @@ public DeltaLakeConfig setRegisterTableProcedureEnabled(boolean registerTablePro
         this.registerTableProcedureEnabled = registerTableProcedureEnabled;
         return this;
     }
+
+    public int getReaderVersion()
+    {
+        return readerVersion;
+    }
+
+    @Config("delta.reader-version")
+    @ConfigDescription("The reader version used by tables created by Trino")
+    public DeltaLakeConfig setReaderVersion(int readerVersion)
+    {
+        this.readerVersion = readerVersion;
+        return this;
+    }
+
+    public int getWriterVersion()
+    {
+        return writerVersion;
+    }
+
+    @Config("delta.writer-version")
+    @ConfigDescription("The writer version used by tables created by Trino")
+    public DeltaLakeConfig setWriterVersion(int writerVersion)
+    {
+        this.writerVersion = writerVersion;
+        return this;
+    }
+
+    @AssertTrue(message = "Unsupported reader version for Delta Lake connector")
+    public boolean isValidReaderVersion()
+    {
+        return isValidVersion("delta.reader-version", readerVersion, MIN_READER_VERSION, MAX_READER_VERSION);
+    }
+
+    @AssertTrue(message = "Unsupported writer version for Delta Lake connector")
+    public boolean isValidWriterVersion()
+    {
+        return isValidVersion("delta.writer-version", writerVersion, MIN_WRITER_VERSION, MAX_WRITER_VERSION);
+    }
```

Comment:
Are those functions used only in tests? If yes then they should be probably moved there, or maybe they should be used in production code?

Reply:
Ok. This annotation is used during object validation. It is clear after test changes.